application/shared-kernel/ApiCore/Endpoints/TrackEndpoints.cs

```diff
@@ -0,0 +1,206 @@
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.ApplicationInsights.Extensibility.Implementation;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Routing;
+using StackFrame = Microsoft.ApplicationInsights.DataContracts.StackFrame;
+
+namespace PlatformPlatform.SharedKernel.ApiCore.Endpoints;
+
+public static class TrackEndpoints
+{
+    public static void MapTrackEndpoints(this IEndpointRouteBuilder routes)
+    {
+        routes.MapPost("/api/track", Track);
+    }
+
+    private static TrackResponseSuccessDto Track(
+        List<TrackRequestDto> trackRequests,
+        TelemetryClient telemetryClient,
+        ILogger<string> logger
+    )
+    {
+        foreach (var trackRequestDto in trackRequests)
+        {
+            switch (trackRequestDto.Data.BaseType)
+            {
+                case "PageviewData":
+                {
```

Comment:
Just noticed these `{...}` around the case content. While this is valid syntax it's not normal. I tried to remove them but realize it's because we reuse the `telemetry` name variable, and the braces help us do that. But I think it would be more normal C# if we did `var pageViewTelemetry = new PageViewTelemetry` and removed the braces.